sdk/src/main/java/io/dapr/actors/runtime/ReminderInfo.java

```diff
@@ -0,0 +1,101 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package io.dapr.actors.runtime;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.type.MapType;
+import java.io.IOException;
+import java.time.*;
+import java.util.Base64;
+import java.util.Map;
+
+class ReminderInfo
+{
+  private final Duration minTimePeriod = Duration.ofMillis(-1);
+
+  public Duration dueTime;
+  public Duration period;
+  public byte[] data;
+
+  public ReminderInfo() {
+  }
+
+  public ReminderInfo(byte[] state, Duration dueTime, Duration period) {
+    this.ValidateDueTime("DueTime", dueTime);
+    this.ValidatePeriod("Period", period);
+    this.data = state;
+    this.dueTime = dueTime;
+    this.period = period;
+  }
+
+  Duration getDueTime() {
+    return this.dueTime;
+  }
+
+  Duration getPeriod() {
+    return this.period;
+  }
+
+  byte[] getData() {
+    return this.data;
+  }
+
+  String serialize() throws IOException {
+    try {
+      ObjectMapper om = new ObjectMapper();
+      ObjectNode objectNode = om.createObjectNode();
+      objectNode.put("dueTime", ConverterUtils.ConvertDurationToDaprFormat(this.dueTime));
+      objectNode.put("period", ConverterUtils.ConvertDurationToDaprFormat(this.period));
+      if (this.data != null) {
+        objectNode.put("data", Base64.getEncoder().encodeToString(this.data));
+      }
+
+      return om.writeValueAsString(objectNode);
+    } catch (IOException e) {
+      throw e;
+    }
+  }
+
+  static ReminderInfo deserialize(byte[] stream) throws IOException {
+    try {
```

Comment:
Can we change this desrialization to be more performant by directly iterating over json and also not creating these extra objects like Map etc.
https://github.com/FasterXML/jackson-docs/wiki/Presentation:-Jackson-Performance#compatible-not-so-easy-use-the-streaming-api

It's more code but more performant, do this both for serialization and deserialization.

Reply:
The serialize and deserialize should get using the same type, not String vs byte. Let me fix this since Leon is OOO. Keep the PR open so I remember.

Reply:
I actually need this class now, so I am merging this.